File: dataset/javascript/original/path_traversal.js
const express = require('express');
const path = require('path');
const app = express();

const UPLOAD_DIR = '/var/www/uploads';

app.get('/download', (req, res) => {
  const filename = req.query.file;
  const filepath = path.join(UPLOAD_DIR, filename);
  res.sendFile(filepath);
});

File: dataset/python/original/path_traversal.py
import os
from flask import Flask, request, send_file

app = Flask(__name__)

UPLOAD_DIR = "/var/www/uploads"

@app.route("/download")
def download_file():
    filename = request.args.get("file")
    filepath = os.path.join(UPLOAD_DIR, filename)
    return send_file(filepath)
